test/Microsoft.AspNetCore.SignalR.Client.Tests/TestHttpMessageHandler.cs

```diff
@@ -28,20 +29,28 @@ public TestHttpMessageHandler(bool autoNegotiate = true)
             return await _handler(request, cancellationToken);
         }
 
-        public static HttpMessageHandler CreateDefault()
+        public static TestHttpMessageHandler CreateDefault(bool handleSend = true)
         {
             var testHttpMessageHandler = new TestHttpMessageHandler();
 
-            testHttpMessageHandler.OnSocketSend((_, __) => ResponseUtils.CreateResponse(HttpStatusCode.Accepted));
+            if (handleSend)
+            {
+                testHttpMessageHandler.OnSocketSend((_, __) => ResponseUtils.CreateResponse(HttpStatusCode.Accepted));
+            }
             testHttpMessageHandler.OnLongPoll(async cancellationToken =>
             {
-                // Just block until canceled
-                var tcs = new TaskCompletionSource<object>();
-                using (cancellationToken.Register(() => tcs.TrySetResult(null)))
+                if (!testHttpMessageHandler.IsFirstRequest)
                 {
-                    await tcs.Task;
+                    // Just block until canceled
+                    var tcs = new TaskCompletionSource<object>();
+                    using (cancellationToken.Register(() => tcs.TrySetResult(null)))
+                    {
+                        await tcs.Task;
+                    }
+                    return ResponseUtils.CreateResponse(HttpStatusCode.NoContent);
                 }
-                return ResponseUtils.CreateResponse(HttpStatusCode.NoContent);
+                testHttpMessageHandler.IsFirstRequest = false;
```

Comment:
Comment: "Otherwise, if this is the first request, just toggle the first request flag and send an empty response"